[CS2103T-F14-4] Deliveria

tysg opened the pull request:
@ArgVampir 
@junnbang
@cheeyou
@SamYJ2606

tysg committed:
Update the master branch to the upstream repo (#57)

Krit-K committed:
Re-edited readme

SamYJ2606 committed:
test

cheeyou committed:
Added intro part of UG

tysg committed:
Add 3.8-3.11 and FAQ

junnbang committed:
Updated "Getting Started" in User guide.

tysg committed:
Merge pull request #63 from cheeyou/addUserGuide

Added intro part of UG

SamYJ2606 committed:
UG-Draft

junnbang committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F14-4/main into update-userguide

# Conflicts:
#	docs/UserGuide.adoc

tysg committed:
Change contact-us and add Makefile

tysg committed:
Update UserGuide.adoc

tysg committed:
Merge pull request #64 from junnbang/update-userguide

Updated "Getting Started" in User guide.

Krit-K committed:
Merge branch 'master' into Feature3-3.3

tysg committed:
Merge pull request #60 from ArgVampir/Feature3-3.3

Edited section Feature 3 - 3.3

Commit:
Merge branch 'master' into test

tysg committed:
Merge pull request #61 from SamYJ2606/test

UG-Add 3.4 to 3.7

junnbang committed:
Updated "About Us".

tysg committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F14-4/main into ty/edit-ug

tysg committed:
Edit Readme doc

tysg committed:
Commit

Commit:
Merge pull request #62 from tysng/ty/edit-ug

UG: Add 3.8-3.11 and FAQ

tysg committed:
Delete gibberish

tysg committed:
Merge pull request #66 from tysng/ty/edit-ug

Delete gibberish

Krit-K committed:
Edit User Stories table (Last 3 rows)

junnbang committed:
Updated "About Us".

junnbang committed:
Merge branch 'master' of https://github.com/junnbang/main into update-about-us

tysg committed:
Merge pull request #67 from ArgVampir/UG-last3

Edit User Stories table (Last 3 rows)

tysg committed:
Merge pull request #65 from junnbang/update-about-us

Updated "About Us".

junnbang committed:
Updated `use cases` section in DeveloperGuide.

junnbang committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F14-4/main into updated-developerguide

junnbang committed:
Fixed "About US" picture not showing.

junnbang committed:
Updated developerguide (#68)

* Updated `use cases` section in DeveloperGuide.

* Fixed "About US" picture not showing.

junnbang committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F14-4/main into updated-developerguide

junnbang committed:
Fixed "AboutUs" image's relative path.

Krit-K committed:
Merge pull request #69 from junnbang/updated-developerguide

Fixed "AboutUs" image relative path.

SamYJ2606 committed:
DG-ProductScope+Picture

Commit:
Merge pull request #71 from SamYJ2606/DG+Picture

DG-ProductScope+Picture

Krit-K committed:
added image (#70)

tysg committed:
Add developer photo

tysg committed:
Update NFR and Glossary

tysg committed:
Merge pull request #74 from tysng/ty/dg-image

Update DG and photo

cheeyou committed:
Added user stories (#73)

* Added User Stories

* Added image for About Us

* Update developer guide

Updated our product name and team name.

Krit-K committed:
Resolve Picture not showing (#75)

* added image

* remove image

* argvampir.png

Commit:
ImageDelete

Commit:
test

Commit:
CaseChange

Commit:
Merge pull request #76 from SamYJ2606/NewImage

Image not showing Resolve

ybchen97 commented on docs/DeveloperGuide.adoc:
```diff
@@ -337,26 +337,73 @@ Use case ends.
 * 3a. The given index is invalid.
```
Hmm, but what if the driver has an ongoing delivery task?

olivercheok20 commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +295,35 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+|`* * *` |driver manager |sort and display delivery tasks by their starting time |view pending tasks in an orderly manner
 
-|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
+|`* * *` |driver Manager |search for tasks by a keyword |find a task more easily
```
Inconsistent capitalisation.

olivercheok20 commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +295,35 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+|`* * *` |driver Manager |search for tasks by a keyword |find a task more easily
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+|`* *` |driver Manager |toggle dark or light theme for the interface | the UI can change according to user preference.
```
Inconsistent punctuation.

ybchen97 commented on docs/DeveloperGuide.adoc:
```diff
@@ -337,26 +337,73 @@ Use case ends.
+
+1. Delivery Manager view the incomplete task list.
+2. Deliveria shows the incomplete task list.
+3. User assign a driver to one of the task
```
What if the driver list or task list is empty?

olivercheok20 commented on docs/DeveloperGuide.adoc:
```diff
@@ -279,13 +279,13 @@ Refer to the guide <<DevOps#, here>>.
+*Value proposition*: manage delivery tasks faster than a typical mouse/GUI driven app
 
 [appendix]
 == User Stories
```
1. Missing the user stories in which the user wants to create, update, or delete deliveries.
2. Perhaps a nice user story would be to prompt the user when there are insufficient drivers to accommodate all deliveries within a certain timeframe.

olivercheok20 commented:
👍

ybchen97 reviewed (commented):
Looks like a great app! I think the use cases can be elaborated a little more :). Maybe you could consider displaying completed deliveries for the user?

olivercheok20 commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +295,35 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+|`*` |driver Manager |see the image of the drivers |can identify them easily
 |=======================================================================
 
 _{More to be added}_
```
Looking forward to it :)

olivercheok20 reviewed (commented):
👍

junnbang committed:
Added `Task`, `TaskList`, `TaskManager`
-TaskList - manage Task
-TaskManager - manage TaskList

Krit-K committed:
Merge pull request #81 from junnbang/add-task

Adds `Task`, `TaskList`, `TaskManager` classes

junnbang committed:
Update About Us with draft portfolios. (#82)

junnbang committed:
Revert "Adds `Task`, `TaskList`, `TaskManager` classes" (#83)

tysg committed:
Add schedule and duration class (#80)

tysg committed:
Add IDE config to pass stylecheck (#79)

Commit:
Customer + CustomerList

Commit:
PassCheckstyle

Krit-K committed:
Merge pull request #86 from SamYJ2606/AddCustomerplusCustomerList

Add Customer and CustomerList

Krit-K committed:
add Goods and Description (#78)

junnbang committed:
Adds `Tasks, `TaskList`, `TaskManager`. (#85)

tysg committed:
Support finding first avaiable time slot (#92)

cheeyou committed:
Add driver (#87)

Added Driver, DriverList and DriverManager.

junnbang committed:
Add task commands (#91)

Added Task Command for add and delete.

tysg committed:
Refractor AddressBook to generic (#95)

junnbang committed:
Edit task command (#100)

Added EditTaskCommand and updated relevant managers.

tysg committed:
Add assign command (#101)

junnbang committed:
Updated JsonAdaptedTask, DeleteIdCommand(for task), EditTaskCommand (#102)

tysg committed:
Add free command (#106)

Add unassign feature

junnbang committed:
Update UG and DG. (#107)

tysg committed:
Update Assign command (#108)

tysg committed:
Update Model UML diagram (#110)

Commit:
[RESET MASTER] Add customer, driver, and managers (#98)

cheeyou committed:
Edit and find customers and drivers (#109)

Added EditCommand for Customer and Drivers.
Added FindCommand for Customers and Drivers.

Commit:
Update UG&DG (#112)

ybchen97 commented on docs/DeveloperGuide.adoc:
```diff
@@ -262,6 +309,82 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+It uses the `AddCustomerCommand`, which extends `Command`, to add a `Customer` into the `CustomerManager`.
+`AddCustomerCommandParser` is also utilised to parse and validate the user inputs before sending it to `AddCustomerCommand` to execute.
+'AddCustomerCommand' requires the following fields: `Customer`.
+The attributes of Task is as follows:
```
Is this a typo? Are you referring to Customer instead of Task?

ybchen97 reviewed (commented):
Hello, good work on the dev guide so far! Some general comments:

1. Add delivery task
- For the sequence diagram, method calls can be written in simple prose as opposed to writing out the actual code. This way each method call can be shorter.
- Missing constructor label for the instantiation of AddTaskCommandParser
- Task class should be in Model instead of Logic
- Diagram is a bit small

2. Customer Feature
- Diagram is a little too small
- Customer extends Person is not reflected in the class diagram

3. Driver Feature
- Driver extends Person is not reflected in the class diagram

Jiayou y'all are almost done!

brianyenna commented on docs/DeveloperGuide.adoc:
```diff
@@ -122,27 +122,26 @@ image::ModelClassDiagram.png[]
 .Structure of the Storage Component
 image::StorageClassDiagram.png[]
+.Structure of the CentralManager
+image::CentralManager.png[]
```
The CentralManager is by itself on the top right hand corner and s not connected to anything else in the diagram.
![Screen Shot 2019-10-25 at 11 36 06 AM](https://user-images.githubusercontent.com/37840339/67541542-dd456100-f71b-11e9-8bf9-b29f7881b526.png)

hcwong commented on docs/diagrams/AddTaskCommand.puml:
```diff
@@ -0,0 +1,104 @@
+@startuml
+!include style.puml
```
this is very detailed, but maybe you could magnify it a bit cos have to zoom in to see the details

hcwong replied:
This applies to the rest of the sequence diagrams.

hcwong commented on docs/DeveloperGuide.adoc:
```diff
@@ -153,6 +152,54 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+==== Implementation
+
+Every `Driver` keeps track of a `Schedule` class, which is backed by a naturally sorted, TreeSet of `EventTime` objects.
```
Very nice description

brianyenna commented on docs/DeveloperGuide.adoc:
```diff
@@ -122,27 +122,26 @@ image::ModelClassDiagram.png[]
 The `Model`,
```
Do you think that the multiplicity should be labelled for Task? For example, how many Drivers should there be for a single Task. 

Also, do you think that the multiplicity from ModelManager to Person should be * instead of 1? The label "filtered list" might be inaccurate for a diagram. Do you think that the label "filtered list" could be changed to a role instead? For example, "<| filteredPersonsList". 

ModelManager also has a FilteredList of Drivers and Customers. Do you think that these could be added into your diagram as well?

hcwong reviewed (commented):
Good job just continue to update in the future 👍

brianyenna commented on docs/DeveloperGuide.adoc:
```diff
@@ -262,6 +309,82 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+
+The following sequence diagram shows how the add driver operation works:
+
+.Sequence Diagram of adding a task.
```
Good looking sequence diagram :)

brianyenna reviewed (commented):
Good looking DG :)

junnbang committed:
Idmanager and refractor code. (#113)

* Added IdManager. To be used for storage of Last Unique Id used by Task,Customer and Driver.

* Added JsonAdaptedDriver

* Added JsonAdaptedDriver.

* Refractor AddCommand and EditCommand for Driver and Customer to take in unique id.

* Updated DeleteIdCommand, setDriver method in Task.
-setDriver in Task will now adjust Status and EventTime accordingly.
-DeleteIdCommand will now prevent deletion of driver or customer who are already allocated to a task.

* Updated EditTaskCommand.
-if edited date is before today, then don't allow.

* Updated after review.

junnbang committed:
Centralized Storage for all managers. (#118)

junnbang committed:
Added Done Command (Only can change ONGOING task to COMPLETED). (#121)

junnbang committed:
Added Read Command  (#120)

* Added Read Command (Get details of task / customer / driver) base on its unique id.

* Abstracted getNoOfPrefixesPresent method to ParseUtil.

* Updated EditTaskCommand

Krit-K committed:
Add ui feature - viewD unfinished (#115)

* added a skeleton of view DriverTasks command

* extra spacing

* fixed bugs

* added TaskList onto the right of the Ui

* Updated CustomerList and TaskList UI display

* UI update for assigned and unassigned task

* fix test driver and customer sample

* Fix unassigned and assigned tasklist taking the same list ID

* Rename taskList to AssignedTaskList
Added Headings for each list
Added description under each lists

* fixed display layout example for assign
Use driver Index instead of the deprecated Index class

* deleted sample and reset back before merge

* remove redundant set tasks

* Added toString for TaskList.

* Fixed tasks not showing in UI.

* update checkstyle.

* Added Send button under the commandbox

junnbang committed:
update UG / DG (#124)

junnbang committed:
Added automated testing for tasks and tasklist. (#125)

junnbang committed:
Added List and Clear Command. Fixed bugs mentioned in issues. (#137)

* Updated EditTaskCommand to fix replace status problem.

* Added EditC,EditD, findC,findD into addressbook parser. Removed those unnecessary commands.

* Updated UG guide link for help command

* Fixed the window size to MAX. Disabled resize window button. Changed name of app to Deliveria.

* Added List Command which reset the task,driver and customer list to its original predicate.

* Removed unnecessary commands tests

* Added Clear command and updated clear command test.

* Updated clear command usage to indicate `CONFIRM` as case sensitive.

tysg committed:
Fix #127: suggested time is no longer in the past (#141)

Krit-K committed:
Fix card and format list (#139)

* added a skeleton of view DriverTasks command

* extra spacing

* fixed bugs

* added TaskList onto the right of the Ui

* Updated CustomerList and TaskList UI display

* UI update for assigned and unassigned task

* fix test driver and customer sample

* Fix unassigned and assigned tasklist taking the same list ID

* Rename taskList to AssignedTaskList
Added Headings for each list
Added description under each lists

* fixed display layout example for assign
Use driver Index instead of the deprecated Index class

* deleted sample and reset back before merge

* remove redundant set tasks

* Added toString for TaskList.

* Fixed tasks not showing in UI.

* update checkstyle.

* Added Send button under the commandbox

* Re-format Driver List, Unassigned and Assigned TaskLists and CustomerList

* fix checkstyle issues

* New layout and removed Assigned: for Incomplete Delivery task list

* Remove person list and update check style

tysg committed:
Add PPP (#147)

tysg committed:
Release admin fixes

tysg committed:
Fix UI not updating and change free command syntax (#151)

tysg committed:
Fix suggested time cannot be added and assign time format issues (#154)

junnbang committed:
Added savepdf Command. (#208)

* Added skeleton of the savepdf command.

* Added initial pdf creator

* Fixed mispelled file directory.

* Added savePdf command.

* sort tasks by eventTime

* Format pdf

* Updated Checkstyle.

* updated after review

junnbang committed:
Fix bugs. Removed unused commands from UG. Added refresh function for list. (#210)

* Updated UG. Removed all unused commands.

* Added refresh functions for all list. Fixed the bugs of not refreshing.

* Minor fix on editT

* minor changes to pdf command

tysg committed:
Restrict assign command to today (#215)

tysg committed:
Fix user guide ui image (#213)

tysg committed:
Fix documentation and command descriptions (#212)

tysg committed:
Fix driver display (#214)

junnbang committed:
added badges for code quality and coverage (#218)

Commit:
Fix #136 : Clear Schedule at end of the day (#143)

tysg committed:
Enhance assign workflow (#219)

junnbang committed:
Update DG for savepdf, some refractoring for savepdf stuffs, fix bugs (#220)

Commit:
Bugfix for Notification window Issue #222 and #206 (#223)

tysg committed:
Improve error message (#228)

tysg committed:
Fix: Schedule is suggesting a time that conflicts with the existing schedule (#227)

tysg committed:
Improve schedule presentation (#229)

junnbang committed:
added automated testing for some commands (#230)

* added SavePdfCommandTest and DoneCommandTest

* added EditTaskCommandTest

Krit-K committed:
View completed tasks (#224)

added viewC, viewD commands

tysg committed:
Improve abstraction (#231)

junnbang committed:
updated UG, savepdf command and update command

junnbang committed:
updated DG.

junnbang committed:
updated ug edittask command

Commit:
Added Driver Rating System #52 View customer and driver window #241 (#234)

* Added Driver Rating System & modify Done command to have a rating when done command is used

* pass checkstyle

* added import

* made relevant changes

* Resolved comments and added viewWindow Feature[WIP]

* added working fxml files

* added working map in customerwindow

* truncated toString()

* edited Customer and Driver Cards to show just essential details

* edited name of windows

* changes made

* checkstyle changes

* edited comments

junnbang committed:
update PPP enhancements

junnbang committed:
Added update command to update Company information and added savepdf for delivery order. (#239)

* Added Company, and JsonCompany for storage

* integrate Company into model and central manager

* added updatecommand to update company information

* Allow GstRegNo to be empty in user input.

* updateed savepdf command to accoomodate delivery order and task summary

* Created the foundation for pdfManager to manage pdf summary and deliver order

* Updated savePdfCommand

* added Company canvas

* Added customer information canvas

* Added delivery order information layout.

* added task description layout.

* added terms and conditions layout

* added customer signature layout

* added company signature layout

* Updated savepdf test

* moved pdf files to another folder.

* update on pdf layout

junnbang committed:
updated PPP

junnbang committed:
Fix editC/editD command not updating task information. Change "AddressBookParser" to "CommandParser" (#245)

junnbang committed:
update DG.

tysg committed:
Fix parser (#243)

Krit-K committed:
History and statistic (#246)

* added a skeleton of view DriverTasks command

* extra spacing

* fixed bugs

* added TaskList onto the right of the Ui

* Updated CustomerList and TaskList UI display

* UI update for assigned and unassigned task

* fix test driver and customer sample

* Fix unassigned and assigned tasklist taking the same list ID

* Rename taskList to AssignedTaskList
Added Headings for each list
Added description under each lists

* fixed display layout example for assign
Use driver Index instead of the deprecated Index class

* deleted sample and reset back before merge

* remove redundant set tasks

* Added toString for TaskList.

* Fixed tasks not showing in UI.

* update checkstyle.

* Added Send button under the commandbox

* Re-format Driver List, Unassigned and Assigned TaskLists and CustomerList

* fix checkstyle issues

* New layout and removed Assigned: for Incomplete Delivery task list

* Remove person list and update check style

* added Completed TaskListPanel and rename TaskList to AssignedTaskList

* created completed taskList panel

* change command format from viewC to viewC c/[index number]

* update the parser for driver and customer find command

* added updateCompleteTask function

* remove unnecessary imports

* Completed navigation command and fix view driver/customer command

* fix list to update completed task also

* removed a line

* fixed checkstyle

* removed white space

* changes all class to use one static tab name, remove unused methods, update javadocs

* added observable command list

* add show all commands predicate and update statistic tab

* Added 2 bar graphs for customer and driver statistics

* add more sample data

* Rename the chart title and fix driver using same ID

* Updated User guide doc

* Added DG and fix bar chart display

* fix Event time displaying the same time

* Changed to use global clock

Commit:
Update UG/DG (#244)

tysg committed:
Reverting a change that overwritten a commit (#247)

junnbang committed:
update PPP

tysg committed:
Update UG, DG and portfolio page (#236)

* Changes

* Edit css according to forum

* Update suggest, assign, free details

* intermediate commit

* Complete Getting started

* Remove html

* Update ug dg graph ppp

* Changes

* Change colour

* Remove links

* Format

* whitespace

* Change Ui

* Changes

* Changes

junnbang committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F14-4/main into update-UG-DG-PPP

# Conflicts:
#	docs/UserGuide.adoc

junnbang committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F14-4/main into update-UG-DG-PPP

# Conflicts:
#	docs/DeveloperGuide.adoc
#	docs/UserGuide.adoc

tysg committed:
Change version (#249)

tysg committed:
Update ug dg ppp (#248)

Krit-K committed:
Ugdg (#250)
